commit dbcf322e61bb239aaa580a766706402ea0a226b0 (refs/changes/54/154/1)
Author: Julien Gilli <julien.gilli@joyent.com>
Date:   2016-07-27T15:16:41-07:00 (3 years, 2 months ago)
    
    TOOLS-1499: add -C channel command line option to sdcadm experimental update-gz-tools

diff --git a/lib/cli/do_update_gz_tools.js b/lib/cli/do_update_gz_tools.js
index 41b51a0..3bc7440 100644
--- a/lib/cli/do_update_gz_tools.js
+++ b/lib/cli/do_update_gz_tools.js
@@ -8,6 +8,8 @@
  * Copyright 2016 Joyent, Inc.
  */
 
+var assert = require('assert-plus');
+
 var errors = require('../errors');
 
 /*
@@ -24,6 +26,9 @@ function do_update_gz_tools(subcmd, opts, args, cb) {
     var progress = self.progress;
     var execStart = Date.now();
 
+    assert.object(opts, 'opts');
+    assert.optionalString(opts.channel, 'opts.channel');
+
     if (opts.help) {
         this.do_help('help', {}, [subcmd], cb);
         return;
@@ -43,6 +48,11 @@ function do_update_gz_tools(subcmd, opts, args, cb) {
             'must specify installer image UUID or --latest'));
     }
 
+    // Set or override the default channel if anything is given:
+    if (opts.channel) {
+        self.sdcadm.updates.channel = opts.channel;
+    }
+
     self.sdcadm.updateGzTools({
         image: opts.latest ? 'latest' : args[0],
         progress: progress,
@@ -80,6 +90,12 @@ do_update_gz_tools.options = [
         help: 'Number of concurrent servers downloading cn_tools file or ' +
             'being updated simultaneously. Default: 5',
         helpArg: 'CONCURRENCY'
+    },
+    {
+        names: ['channel', 'C'],
+        type: 'string',
+        help: 'Use the given channel to fetch the image, even if it is not the '
+            + 'default one.'
     }
 ];
 do_update_gz_tools.help = (
diff --git a/man/man1/sdcadm.1.ronn b/man/man1/sdcadm.1.ronn
index 511d641..baf4998 100644
--- a/man/man1/sdcadm.1.ronn
+++ b/man/man1/sdcadm.1.ronn
@@ -982,6 +982,9 @@ The eventual goal is to integrate all of this into "sdcadm update".
 `--force-reinstall`
     Force reinstall of the current gz-tools image in use.
 
+`-C, --channel`
+    Use the given channel to fetch the image, even if it is not the default one.
+
 ### sdcadm experimental add-new-agent-svcs
 
 Temporary grabbag for installing the SDC global zone new agents.
